### PR TITLE
drivers: pwm: header cleanup

### PR DESCRIPTION
**Pull request by gmarull**
This PR includes some prep-work before introducing `pwm_dt_spec` facilities, as some other APIs do. It mainly contains:

- Doxygen cleanup/improvements
- Code formatting
- Minor code improvements (error handling, includes...)

### CODE REVIEW

**Comment by henrikbrixandersen** [comment]
> I think we should be consistent everywhere, that is, rename the API to be `pwm_channel_...` as well (with `pwm_pin_...` deprecation as it is stable).

I agree. Did you want to include this change + deprecation in this PR?

**Review by henrikbrixandersen — DISMISSED** [review]

**Review by carlescufi — DISMISSED** [review]
The current set of changes is fine with me.

**Comment by gmarull** [comment]
> > I think we should be consistent everywhere, that is, rename the API to be `pwm_channel_...` as well (with `pwm_pin_...` deprecation as it is stable).
> 
> I agree. Did you want to include this change + deprecation in this PR?

I've dropped name changes from this PR, will open a follow-up PR with the proposal.

File: include/drivers/pwm.h [hunk 1]
```diff
@@ -21,10 +21,13 @@
  */
 
 #include <errno.h>
-#include <zephyr/types.h>
-#include <stddef.h>
-#include <sys/math_extras.h>
+#include <stdint.h>
+
+#include <autoconf.h>
```
[the remark this answers is not in the document]
Comment (gmarull) [reply include/drivers/pwm.h:26]: I've just tried to make the header self-contained, I'm not sure if `autoconf.h` is supposed to be part of some other "catch-all" header, do you know?

**Review by carlescufi — DISMISSED** [review]

File: include/drivers/pwm.h [hunk 2]
```diff
@@ -232,7 +231,7 @@ static inline int pwm_pin_configure_capture(const struct device *dev,
 					    pwm_capture_callback_handler_t cb,
 					    void *user_data)
 {
-	const struct pwm_driver_api *api = (struct pwm_driver_api *)dev->api;
+	const struct pwm_driver_api *api = dev->api;
```
Comment (anangl) [inline include/drivers/pwm.h]: These type castings are not redundant. They are needed when this header file is included from a C++ source file.

**Review by anangl — CHANGES_REQUESTED** [review]

File: include/drivers/pwm.h (hunk 2 above)
Comment (gmarull) [reply include/drivers/pwm.h]: Argh, forgot about C++ :-(

**Review by anangl — DISMISSED** [review]

File: include/drivers/pwm.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (gmarull) [reply include/drivers/pwm.h]: ok, figured it out, works using `-imacros`, so it doesn't have to be included. removed

**Review by anangl — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

File: include/drivers/pwm.h [hunk 3]
```diff
@@ -21,10 +21,12 @@
...
 
 #include <errno.h>
-#include <zephyr/types.h>
-#include <stddef.h>
-#include <sys/math_extras.h>
+#include <stdint.h>
+
 #include <device.h>
+#include <sys/math_extras.h>
+#include <toolchain.h>
```
Comment (nashif) [inline include/drivers/pwm.h:28]: what do we need from here?

**Pull request merged**